private/buf/bufcli/errors.go

```diff
@@ -181,10 +181,9 @@ func wrapError(err error) error {
 			}
 			return errors.New(msg)
 		}
-		return fmt.Errorf("Failure: %s", connectErr.Message())
```

Comment:
Looks like we did not print the error code before - it was just:

```
Failure: 403 Forbidden
```

I think this is intentional, see "intuitive message" comment at the top of the block.

With this change, we're adding the helpful message to the output, but we're also adding stutter in the form of "Failure: permission_denied: 403 Forbidden."

I don't see an easy way to retain the error without printing the code here, but I think we should make this possible. Someone more familiar with the code base than me might have an idea.

Reply:
Thanks I missed that, fixed the description. On the error site could avoid the `%w` to not wrap and interpret the connect error message: https://github.com/bufbuild/buf/blob/9bab8d4d668c993085b19636b2f18a6723c30cae/private/buf/bufcli/errors.go#L145
Also not sure if using `NewUnimplementedRemoteError` is the right message for this status code, and other errors on this path aren't wrapped. But for this issue it does make sense.

Reply:
```go
func NewUnimplementedRemoteError(err error, remote string, moduleIdentity string) error {
	if connectErr, ok := asConnectError(err); ok {
		err = connectErr.Unwrap()
	}
	return fmt.Errorf("%w. Are you sure %q (derived from module name %q) is a Buf Schema Registry?", err, remote, moduleIdentity)
}
```

Reply:
Pushed up a fix to always unwrap the connect error. Removes the `permission_denied` code.